testing/internal/e2e/tests/base/target_tcp_connect_redis_test.go

```diff
@@ -11,23 +11,122 @@ import (
 
+	ctx := context.Background()
+	redisInfo := setupRedisContainer(t, ctx)
+
+	cases := []struct {
+		name             string
+		expectedUsername string
+		extraSetup       func(t *testing.T, ctx context.Context, resources *boundaryResources, redisInfo *redisContainerInfo)
+	}{
+		{
+			name:             "UsernamePassword",
+			expectedUsername: redisInfo.Username,
+			extraSetup: func(t *testing.T, ctx context.Context, resources *boundaryResources, redisInfo *redisContainerInfo) {
+				credentialId, err := boundary.CreateStaticCredentialUsernamePasswordCli(
+					t,
+					ctx,
+					resources.storeId,
+					redisInfo.Username,
+					redisInfo.Password,
+				)
+				require.NoError(t, err)
+
+				err = boundary.AddBrokeredCredentialSourceToTargetCli(t, ctx, resources.targetId, credentialId)
+				require.NoError(t, err)
+			},
+		},
+		{
+			name:             "Password",
+			expectedUsername: "default",
+			extraSetup: func(t *testing.T, ctx context.Context, resources *boundaryResources, redisInfo *redisContainerInfo) {
+				credentialId, err := boundary.CreateStaticCredentialPasswordCli(
+					t,
+					ctx,
+					resources.storeId,
+					redisInfo.Password,
+				)
+				require.NoError(t, err)
+
+				err = boundary.AddBrokeredCredentialSourceToTargetCli(t, ctx, resources.targetId, credentialId)
+				require.NoError(t, err)
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			// Setup
+			resources := setupBoundaryCredentialStore(t, ctx, redisInfo.Hostname, redisInfo.Port)
+			c.extraSetup(t, ctx, resources, redisInfo)
+
+			// Flush redis between tests
+			t.Cleanup(func() {
+				flushRedis(t, ctx, resources.targetId)
+			})
+
+			// Validation
+			cmd := exec.CommandContext(ctx,
+				"boundary",
+				"connect", "redis",
+				"-target-id", resources.targetId,
+			)
+
+			stdin, err := cmd.StdinPipe()
+			require.NoError(t, err)
+			stdout, err := cmd.StdoutPipe()
+			require.NoError(t, err)
+			require.NoError(t, cmd.Start())
+
+			output, err := sendRedisCommand(stdin, stdout, "ACL WHOAMI\r\n")
+			require.NoError(t, err)
+			require.Equal(t, c.expectedUsername, output)
+
+			output, err = sendRedisCommand(stdin, stdout, "SET e2etestkey e2etestvalue\r\n")
+			require.NoError(t, err)
+			require.Equal(t, "OK", output)
+
+			output, err = sendRedisCommand(stdin, stdout, "GET e2etestkey\r\n")
+			require.NoError(t, err)
+			require.Equal(t, "e2etestvalue", output)
+
+			output, err = sendRedisCommand(stdin, stdout, "GET e2etestkey\r\n")
+			require.NoError(t, err)
+			require.Equal(t, "e2etestvalue", output)
+
+			output, err = sendRedisCommand(stdin, stdout, "QUIT\r\n")
+			require.Equal(t, io.EOF, err)
+			require.Empty(t, output)
+
+			// Confirm that boundary connect has closed
+			err = cmd.Wait()
+			require.NoError(t, err)
+		})
+	}
+}
+
+// setupRedisContainer starts a Redis container and returns its connection info
+func setupRedisContainer(t *testing.T, ctx context.Context) *redisContainerInfo {
```

Comment:
Feels like some of these redis methods might be better suited in another package? One idea is in `e2e/infra/redis_docker.go`?